tools/nsp_scripts/test_model_script.py

```diff
@@ -15,31 +22,258 @@
+        type=str,
+        help="Directory to pretrained NLU model"
+    )
+    # optimization arugments
+    parser.add_argument("--batch_size", default=28, type=int, help="Batch size")
+    parser.add_argument(
+        "--train_encoder", default=1, type=int, help="Whether to finetune the encoder"
+    )
+    parser.add_argument(
+        "--lambda_span_loss",
+        default=0.5,
+        type=float,
+        help="Weighting between node and span prediction losses",
+    )
+    parser.add_argument(
+        "--node_label_smoothing",
+        default=0.0,
+        type=float,
+        help="Label smoothing for node prediction",
+    )
+    parser.add_argument(
+        "--dtype_samples",
+        default="templated:.55;templated_filters:.05;annotated:.4",
+        type=str,
+        help="Sampling probabilities for handling different data types",
+    )
+    parser.add_argument(
+        "--examples_per_epoch", default=-1, type=int, help="Number of training examples per epoch"
+    )
+    parser.add_argument(
+        "--rephrase_proba", default=-1.0, type=float, help="Only specify probablility of rephrases"
+    )
+    parser.add_argument(
+        "--show_samples", action="store_true", help="show samples every few iterations"
+    )
+    # debug arguments
+    parser.add_argument(
+        "--vis_step_size", default=10, type=int, help="The number of iterations to visualize chat and parsed tree"
+    )
+    parser.add_argument(
+        "--noop_thres", default=0.95, type=float, help="The threshold of NOOP action"
+    )
+    parser.add_argument(
+        "--beam_size", default=5, type=int, help="Number of branches to keep in beam search"
+    )
+    parser.add_argument(
+        "--well_formed_pen", default=1e2, type=float, help="Penalization for poorly formed trees"
+    )
+    args = parser.parse_args()
+    dtype_samples = {}
+    for x in args.dtype_samples.split(";"):
+        y = x.split(":")
+        dtype_samples[y[0]] = float(y[1])
+    args.dtype_samples = dtype_samples
+
+    os.makedirs(args.output_dir, exist_ok=True)
+    # HACK: allows us to give rephrase proba only instead of full dictionary
+    if args.rephrase_proba > 0:
+        args.dtype_samples = json.dumps(
+            [["templated", 1.0 - args.rephrase_proba], ["rephrases", args.rephrase_proba]]
+        )
+    
+    # load grammar tree
+    if isfile(args.tree_voc_file):
+        logging.info("====== Loading Grammar ======")
     else:
-        tree = btr[0][0]
-    return tree
+        logging.info("====== Making Grammar ======")
+        build_grammar(args)
+    with open(args.tree_voc_file) as fd:
+        full_tree, tree_i2w = json.load(fd)
+    full_tree_voc = (full_tree, tree_i2w)
+
+    logging.info("====== Loading Pretrained Parameters ======")
+    sd, _, _, _, _ = load_model(args.model_dir)
+    logging.info("====== Setting up Model ======")
+    decoder_with_loss, encoder_decoder, tokenizer = build_model(args, full_tree_voc[1])
```

Comment:
nit: this looks like a file from a different PR ?

Reply:
@kavyasrinet Yeah. It seems somehow I updated this script with the change in this [PR](https://github.com/facebookresearch/fairo/pull/1201). Should I change it back to the script of main branch?

Reply:
We should merge your other PR first. Once that's merged in and we rebase this PR on main, this should go away. But ideally yes, every PR should be against main branch and shouldn't have changes from the other PRs

Reply:
Gotcha.